core/commands/ProjectUsersGetCommand.php

```diff
@@ -0,0 +1,188 @@
+<?php
+# MantisBT - A PHP based bugtracking system
+
+# MantisBT is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MantisBT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+use Mantis\Exceptions\ClientException;
+
+require_api( 'authentication_api.php' );
+require_api( 'user_pref_api.php' );
+
+require_once( dirname( __FILE__ ) . '/../../api/soap/mc_account_api.php' );
+require_once( dirname( __FILE__ ) . '/../../api/soap/mc_enum_api.php' );
+
+/**
+ * Sample:
+ * {
+ *   "query": {
+ *     'id'        => 1,
+ *     'page'      => 1,
+ *     'page_size' => 50,
+ *     'access_level' => 1,
+ *     'include_access_levels' => 1
+ *   }
+ * }
+ */
+
+/**
+ * A command to get the users within a project with the specified access level
+ * or higher.
+ */
+class ProjectUsersGetCommand extends Command {
+	/**
+	 * The project id
+	 *
+	 * @var integer
+	 */
+	private $project_id;
+
+	/**
+	 * The minimum access level, users with access level greater or equal to this access level
+	 * will be returned.
+	 *
+	 * @var integer
+	 */
+	private $access_level;
+
+	/**
+	 * The page number (starts with 1)
+	 *
+	 * @var integer
+	 */
+	private $page;
+
+	/**
+	 * The number of users to return per page.
+	 *
+	 * @var integer
+	 */
+	private $page_size;
+
+	/**
+	 * Include effective access level of users on the project?
+	 *
+	 * @var integer
+	 */
+	private $include_access_levels;
+
+	/**
+	 * The authenticated user id
+	 *
+	 * @var integer
+	 */
+	private $user_id;
+
+	/**
+	 * Constructor
+	 *
+	 * @param array $p_data The command data.
+	 */
+	function __construct( array $p_data ) {
+		parent::__construct( $p_data );
+	}
+
+	/**
+	 * Validate the data.
+	 */
+	function validate() {
+		$this->user_id = auth_get_current_user_id();
+		$this->project_id = (int)$this->query( 'id' );
+		$this->access_level = (int)$this->query( 'access_level' );
+		$this->page = (int)$this->query( 'page', 1 );
+		$this->page_size = (int)$this->query( 'page_size', 50 );
+		$this->include_access_levels = (int)$this->query( 'include_access_levels', true );
+
+		if( $this->project_id <= ALL_PROJECTS ) {
+			throw new ClientException(
+				sprintf( "Must specify a specific project id.", $this->project_id ),
```

Comment:
Should be something like
`sprintf( "Must specify a specific project id, %d is invalid", $this->project_id ),`